### PR TITLE
MULE-18442: Avoid creation of DefaultPolicyManager for each applied P…

### PR DESCRIPTION
**Pull request by balbifm**
…olicy

### CODE REVIEW

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/ApplicationPolicyDeploymentTestCase.java [hunk 1]
```diff
@@ -83,6 +90,12 @@
 
   private static File simpleExtensionJarFile;
   private static File withErrorDeclarationExtensionJarFile;
+
+  @Parameterized.Parameters(name = "Parallel: {0}")
+  public static List<Boolean> params() {
```
Comment (elrodro83) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/ApplicationPolicyDeploymentTestCase.java:95]: remove this

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/ApplicationPolicyDeploymentTestCase.java [hunk 2]
```diff
@@ -119,6 +132,30 @@ public void appliesApplicationPolicy() throws Exception {
     doApplicationPolicyExecutionTest(parameters -> true, 1, POLICY_PROPERTY_VALUE);
   }
 
+  @Test
+  public void applicationPolicyHasNoOpPolicyManagerInjected() throws Exception {
```
Comment (elrodro83) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/ApplicationPolicyDeploymentTestCase.java:136]: `@Issue`

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/ApplicationPolicyRedeploymentTestCase.java [hunk 3]
```diff
@@ -35,6 +39,11 @@
   private static final String OS_POLICY_ID = "object-store-policy";
   public static final String FLOW_NAME = "main";
 
+  @Parameterized.Parameters(name = "Parallel: {0}")
+  public static List<Boolean> params() {
```
Comment (elrodro83) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/ApplicationPolicyRedeploymentTestCase.java:43]: remove

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/ApplicationPolicyDeploymentTestCase.java (hunk 1 above)
Comment (balbifm) [reply modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/ApplicationPolicyDeploymentTestCase.java:95]: Don't think we should remove. We are running policy tests twice for parallel and non-parallel deployment when that does not affect policies at all.

**Comment by balbifm** [comment]
--validate

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/ApplicationPolicyDeploymentTestCase.java (hunk 1 above)
Comment (elrodro83) [reply modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/ApplicationPolicyDeploymentTestCase.java:95]: can you add that as a comment in code?

**Pull request merged**